ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/WSQueryEndpoint.java

```diff
@@ -241,6 +246,17 @@ public void onError(final Session session, final Throwable t) {
     log.error("websocket error in session {}", session.getId(), t);
   }
 
+  private void checkEndpointAuthorization(final Principal userPrincipal, final String methodName) {
+    final Class className = this.getClass();
+
+    if (!securityExtension.getAuthorizer().hasAccess(userPrincipal, className, methodName)) {
+      final String userName = (userPrincipal != null) ? userPrincipal.getName() : null;
+      throw new KsqlException(
+          String.format("User:%s is denied to access this cluster.", userName)
```

Comment:
should this be a 403 exception? how does this error surface to the user?  also I think we should have a different error for null user, and we should have a more descriptive error message:
`"Access denied for user %s to %s:%s".format(username, className, methodName)`

Reply:
Does the user need to know internal class and method names? For them, they just want to open a session to the KSQL server, and the message clarifies they don't have access to it.

Perhaps when we have finer-grained privileges on KSQL, we could mention what resources are denied and why, and what permissions they require.

Reply:
I feel pretty strongly that error messages should be actionable. As it stands, I have no idea what I need to do in order to make progress. Showing the resource/method pairing would help me understand what i need to get